tutorials/elixir-phoenix-on-kubernetes-google-container-engine/index.md

```diff
@@ -268,6 +268,9 @@ Execute the following command to run the build:
 Replace `${PROJECT_ID}` with the ID of your Google Cloud Platform project.
 The period at the end is required.
 
+If your `.gitignore` file is excluding `/config/*.secret.exs`,
+you'll have to comment that out for this step - we need that file.
```

Comment:
I don't think the `.gitignore` file itself is the cause, and commenting out the `prod.secret.exs` entry from `.gitignore` won't solve this problem directly. The problem you observed happens if `config/prod.secret.exs` is completely absent—which could happen if you did a *fresh checkout* of a repository that did not include the `prod.secret.exs` (possibly because of the `.gitignore` entry) and then tried to build from there.

I'm okay with including a note to be careful to ensure that any needed `*.secret.exs` file is present, but I think the note as currently worded is misleading since merely commenting out the entry from `.gitignore` won't have any effect.

Reply:
Again - I'll re-conduct this experiment soon - but I'm pretty sure I had the file there, and it didn't include it until I commented out the line in my `.gitignore` (I know it doesn't make sense, perhaps I also commented it in `.dockerignore`?) .... pending investigation

Reply:
what was the result of your investigation?

Reply:
confirmed - if i include `/config/*.secret.exs` in my `.gitignore` file, I get the following build error:

```
Step 7/17 : COPY . .
 ---> ac7aa6c03741
Step 8/17 : RUN mix do deps.get, deps.compile, compile
 ---> Running in 6dcd7f0aeae6
** (Mix.Config.LoadError) could not load config config/prod.secret.exs
    ** (File.Error) could not read file "/opt/app/config/prod.secret.exs": no such file or directory
    (elixir) lib/file.ex:310: File.read!/1
    (mix) lib/mix/config.ex:189: Mix.Config.read!/2
    (mix) lib/mix/config.ex:233: anonymous fn/3 in Mix.Config.read_wildcard!/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
    (stdlib) erl_eval.erl:878: :erl_eval.expr_list/6
    (stdlib) erl_eval.erl:404: :erl_eval.expr/5
The command '/bin/sh -c mix do deps.get, deps.compile, compile' returned a non-zero code: 1
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: exit status 1
```

Reply:
I found this is a log file which pretty clearly describes why this is true:

```
2018-05-23 21:01:57,867 INFO     root            Using default gcloudignore file:
--------------------------------------------------
# This file specifies files that are *not* uploaded to Google Cloud Platform
# using gcloud. It follows the same syntax as .gitignore, with the addition of
# "#!include" directives (which insert the entries of the given .gitignore-style
# file at that point).
#
# For more information, run:
#   $ gcloud topic gcloudignore
#
.gcloudignore
# If you would like to upload your .git directory, .gitignore file or files
# from your .gitignore file, remove the corresponding line
# below:
.git
.gitignore
#!include:.gitignore
```

Reply:
So how should we proceed?

Reply:
Ping, @dazuma, what do you think?